web/src/main/java/com/bazaarvoice/emodb/web/resources/blob/BlobStoreResource1.java

```diff
@@ -88,9 +93,59 @@
     private final BlobStore _blobStore;
     private final Set<String> _approvedContentTypes;
 
+    private final MetricRegistry _metricRegistry;
```

Comment:
Have you looked into doing these metrics with an annotation? It would likely make this more extensible and less verbose.

Reply:
We could also avoid adding the `Subject` to methods where it is not explicitly necessary.

Reply:
> Have you looked into doing these metrics with an annotation? It would likely make this more extensible and less verbose.

we have https://github.com/bazaarvoice/emodb/blob/c63213c05bbc279fa7f45d1f981bb2f9b595a751/common/dropwizard/src/main/java/com/bazaarvoice/emodb/common/dropwizard/metrics/ParameterizedTimed.java that could be extended, or we create new annotation to have parameters as tags and not as part of metric name. also we need to add annotation listener code.
Personally, i prefer to have code explicit and don't think it's worth to spend time on annotation (at least now).



> We could also avoid adding the `Subject` to methods where it is not explicitly necessary.

Not sure i get this. but we have subject id as group key, so we need to have it explicitly in each method.

Reply:
Do you mind if i take a crack at this?

Reply:
as being said, there is no need in this feature.
i would start with creating ticket to refactor this and other metrics with new annotation approach.
go ahead, if you don't have other priorities.

Reply:
@mike-unitskyi @bdevore17 I like all the points discussed here. My two cents - just continue with the all the meters as is. I know its verbose, but it's the pattern we have at various places when there are a bunch of metrics in the same file. I made one comment to reduce the verbosity which may help look wise 😛. We can then create an improvement Jira ticket to tackle this with annotations, and then use that pattern in so many places in the whole project. Hope you both agree with me 😄